src/app/shared/entry/url-deconstruct.pipe.ts

```diff
@@ -0,0 +1,17 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+@ Pipe({
+  name: 'urlDeconstruct'
+})
+export class UrlDeconstructPipe implements PipeTransform {
+
+  transform(providerUrl: string, versionName: string): any {
+    if (providerUrl.includes('github.com') && versionName !== '') {
```

Comment:
You can simply say `&& versionName` -- if the expression is "truthy" (not undefined, not null, not empty), it will pass. This is arguably better, as you code doesn't catch the undefined and null cases, even though they may not be possible when invoked from the HTML in Angular.